File: Core/src/main/java/jokatu/game/input/AnyEventInputAcceptor.java
package jokatu.game.input;

import jokatu.game.event.GameEvent;
import jokatu.game.player.Player;

/**
 * These are {@link AbstractInputAcceptor}s that can fire any {@link GameEvent}.
 */
public abstract class AnyEventInputAcceptor<I extends Input, P extends Player>
		extends AbstractInputAcceptor<I, P, GameEvent> {}

File: Core/src/main/kotlin/jokatu/game/input/AnyEventInputAcceptor.kt
package jokatu.game.input

import jokatu.game.event.GameEvent
import jokatu.game.player.Player
import kotlin.reflect.KClass

/**
 * These are [AbstractInputAcceptor]s that can fire any [GameEvent].
 */
abstract class AnyEventInputAcceptor<I : Input, P : Player>(
		inputClass: KClass<I>,
		playerClass: KClass<P>
) : AbstractInputAcceptor<I, P, GameEvent>(inputClass, playerClass)
